### PR TITLE
Add procotol_vsn to SC fsm + codec

### PR DESCRIPTION
**Pull request by uwiger**
See [PT #164238122](https://www.pivotaltracker.com/story/show/164238122)

Currently, tests have been added to `aesc_fsm_SUITE` verifying that `protocol_vsn` defaults to `1` (the old version), and that message size limits are respected.

Note that there is no way to verify that both sides use the same version. The `protocol_vsn` option would need to be included in the `channel_open` and `channel_accept` messages for this.

### CODE REVIEW

**Comment by lucafavatella** [comment]
I see this PR is a bit dated, I merged it in recent master in a branch and I started additional CI checks:
* [normal CI](https://circleci.com/workflow-run/6a73f05e-b185-4ec6-99f7-bf6c687e11a8)
* [full system tests](https://circleci.com/gh/aeternity/aeternity/11971)

File: apps/aehttp/src/sc_ws_handler.erl [hunk 1]
```diff
@@ -272,6 +272,7 @@ read_channel_options(Params) ->
          Read(<<"responder_amount">>, responder_amount, #{type => integer}),
          Read(<<"channel_reserve">>, channel_reserve, #{type => integer}),
          Read(<<"ttl">>, ttl, #{type => integer, mandatory => false}),
+         Read(<<"protocol_vsn">>, protocol_vsn, #{type => integer, mandatory => false}),
```
Comment (lucafavatella) [inline apps/aehttp/src/sc_ws_handler.erl]: @uwiger Could you please clarify how this relates to [this doc re channel params](https://github.com/aeternity/protocol/blob/3a00b538ed47efa6ab7bb205f07456e06134ff44/node/api/channels_api_usage.md#channel-parameters)? Any docs requiring update?

**Comment by lucafavatella** [comment]
@uwiger It would be great if you included in this PR the system tests of PR #2224 - amending/integrating them.

**Comment by uwiger** [comment]
Closing PR. The solution would have been ok, had the old nodes and the new been compatible at the chain level (specifically for state channels). The solution could perhaps be revived later, if other changes are needed in encode/decode.

**Pull request closed without merging**